schema.graphql

```diff
@@ -2329,3 +2358,157 @@ type CustomClaimType @entity {
   createdBlock: Block!
   updatedBlock: Block!
 }
+
+enum AffirmingPartyEnum {
+  Sender
+  Receiver
+  Mediator
+}
+
+enum AffirmStatusEnum {
+  Affirmed
+  Rejected
+}
+
+enum TransactionStatusEnum {
+  Created
+  Executed
+  Rejected
+}
+
+"""
+Represents the proof of a transaction affirmation
+"""
+type Proof @jsonField {
+  assetId: String!
+  proof: String!
+}
+
+"""
+Represents a confidential account
+"""
+type ConfidentialAccount @entity {
+  id: ID!
+  account: String!
+  creator: Identity!
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Represents a confidential venue
+"""
+type ConfidentialVenue @entity {
+  id: ID!
+  venueId: Int!
+  creator: Identity!
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Used for many to many relation between ConfidentialAsset and ConfidentialVenue
+"""
+type ConfidentialAssetAllowedVenue @entity {
+  id: ID!
+  asset: ConfidentialAsset!
+  venue: ConfidentialVenue!
+  isAllowed: Boolean!
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Represents a confidential asset
+"""
+type ConfidentialAsset @entity {
+  id: ID!
+  assetId: String!
+  ticker: String
+  data: String
+  creator: Identity!
+  auditors: [String!]
+  mediators: [String!]
+  totalSupply: BigInt!
+  venueFiltering: Boolean!
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Represents a confidential asset holder
+"""
+type ConfidentialAssetHolder @entity {
+  id: ID!
+  account: ConfidentialAccount!
+  asset: ConfidentialAsset!
+  amount: String
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Represents a confidential asset transaction affirmation
+"""
+type ConfidentialTransactionAffirmation @entity {
+  id: ID!
+  transaction: ConfidentialTransaction
+  leg: Int
+  account: ConfidentialAccount
+  type: AffirmingPartyEnum
+  status: AffirmStatusEnum
+  proofs: [Proof]
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Represents a leg of a confidential asset transaction
+"""
+type ConfidentialLeg @entity {
+  id: ID!
+  transaction: ConfidentialTransaction!
+  sender: ConfidentialAccount!
+  receiver: ConfidentialAccount!
+  mediators: [String!]
+  auditors: [String!]
+  asset: ConfidentialAsset
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Represents a confidential transaction
+"""
+type ConfidentialTransaction @entity {
+  id: ID!
+  venue: ConfidentialVenue! @index(unique: false)
+  status: TransactionStatusEnum!
+  eventId: EventIdEnum!
+  eventIdx: Int!
+  legs: [ConfidentialLeg!]! @derivedFrom(field: "transaction")
+  affirmations: [ConfidentialTransactionAffirmation] @derivedFrom(field: "transaction")
+  memo: String
+  pendingAffirmations: Int!
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Represents a confidential asset transaction
+"""
+type ConfidentialAssetTransaction @entity {
```

Comment:
```suggestion
type ConfidentialAssetHistory @entity {
```
The name would be a bit confusing to users. Maybe we can add `history` as suffix?